src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingNamedTypeSymbol.cs

```diff
@@ -418,5 +413,17 @@ internal override bool HasInlineArrayAttribute(out int length)
         {
             return _underlyingType.HasInlineArrayAttribute(out length);
         }
+
+#nullable enable
+        internal sealed override bool HasCollectionBuilderAttribute(out TypeSymbol? builderType, out string? methodName)
+        {
+            bool result = _underlyingType.HasCollectionBuilderAttribute(out builderType, out methodName);
+            if (builderType is { })
+            {
+                builderType = this.RetargetingTranslator.Retarget(builderType, RetargetOptions.RetargetPrimitiveTypesByTypeCode);
```

Comment:
pity that `Retarget` doesn't accept null.  It would simplify a lot of codepaths to not have to do the null checks up front.